src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs

```diff
@@ -100,7 +100,8 @@ public async Task AcceptAsync_ClientStartsAndStopsBidirectionStream_ServerAccept
             serverStream.Transport.Input.AdvanceTo(read.Buffer.End);
 
             // Read data from server.
-            var data = await clientStream.ReadAtLeastLengthAsync(TestData.Length).DefaultTimeout();
```

Comment:
I'm using `ReadAtLeastLengthAsync` on purpose. I don't want to read an exact amount. I want stream to give me the expected amount of data but leave the door open to the stream returning more.

There is a bug in HttpClient's response stream that only shows up when you call ReadAsync and ask to read more data that is available.

Reply:
I know. That's not this test though. I kept your test demonstrating the bug in HttpClient's response stream passing in a larger buffer than needed. See [here](https://github.com/dotnet/aspnetcore/pull/35003/files#diff-69bd98bd920bf41ec1df4627369cfb21d6429e4c6eb75255a8b64b5a0d7476a7R217). I don't see the benefit of returning the buffer rather than passing it in. The caller usually has a better idea of the appropriate buffer size and it doesn't really complicate the call sites.